pkg/load/configAgent.go

```diff
@@ -66,18 +70,42 @@ func NewConfigAgent(configPath string, cycle time.Duration, errorMetrics *promet
 		}
 	}, a.cycle)
 
-	// fsnotify reload
-	configWatcher, err := fsnotify.NewWatcher()
-	if err != nil {
-		return nil, fmt.Errorf("Failed to create new watcher: %v", err)
+	reloaderFuncs := []func(ctx context.Context){}
+	if configMapReload {
```

Comment:
Instead of assuming a structure, couldn't we handle both cases with a recursive traversal that considers the full subtree handled if it finds a `..data` element?